src/features/unified-share-modal/HeaderTitleSection.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Classification from '../classification';
+import messages from './messages';
+import type { item as ItemType } from './flowTypes';
+
+type Props = {
+    classification: ClassificationInfo,
+    isEmailLinkSectionExpanded: boolean,
+    item: ItemType,
+    showCollaboratorList: boolean,
+};
+
+const renderModalTitle = (isEmailLinkSectionExpanded, showCollaboratorList, item) => {
+    const { name } = item;
+    let title;
+
+    if (isEmailLinkSectionExpanded) {
+        title = (
+            <FormattedMessage
+                {...messages.emailModalTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    } else if (showCollaboratorList) {
+        title = (
+            <FormattedMessage
+                {...messages.collaboratorListTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    } else {
+        title = (
+            <FormattedMessage
+                {...messages.modalTitle}
+                values={{
+                    itemName: name,
+                }}
+            />
+        );
+    }
+
+    return title;
+};
+
+const HeaderTitleSection = ({ classification, isEmailLinkSectionExpanded, showCollaboratorList, item }: Props) => {
```

Comment:
Can we just call this (and file name) `HeaderTitle` since suffixing it with Section doesn't add much value.

Reply:
thanks, will update!